src/Compilers/Core/Portable/Compilation/CompilationOptions.cs

```diff
@@ -261,6 +261,8 @@ protected set
 
         private readonly Lazy<ImmutableArray<Diagnostic>> _lazyErrors;
 
+        private int _hashCode;
```

Comment:
Any particular reason you're using 0 as the sentinel, not making this `int?`? Just easier for multithreading purposes?

Reply:
force of habit.  it's just a pattern i got very accustomed to for some reason over my years :)  happy to change to nullable if you'd like :)

Reply:
it's also likely that i trust ints more (with nullable, not sure what the multithreading concerns may be) :)

Reply:
It's fine to leave is as, was more for understanding why this approach was taken.

Reply:
I think an `int` won't tear but an `int?` might.

If we're concerned about multithreading, is there a need for some kind of memory barrier when we write to `_hashCode`?

Reply:
I don't believe that there's anything we need to be concerned about with `int`. Reads and writes are atomic, and since the calculation is stable, the worst that could happen is potentially multiple writes of the same value.

Reply:
yup.  i like int because it's just so safe.